maestro-driver-ios/maestro-driver-iosUITests/Routes/Handlers/ScreenDiffHandler.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import XCTest
+import CryptoKit
+import FlyingFox
+import os
+
+class ScreenDiffHandler : RouteHandler {
+    private let logger = Logger(subsystem: Bundle.main.bundleIdentifier!, category: "ScreenDiffHandler")
+    
+    func handle(request: FlyingFox.HTTPRequest) async throws -> FlyingFox.HTTPResponse {
+        let screenshot1 = XCUIScreen.main.screenshot()
+        let screenshot2 = XCUIScreen.main.screenshot()
+        let hash1 = SHA256.hash(data: screenshot1.pngRepresentation)
+        let hash2 = SHA256.hash(data: screenshot2.pngRepresentation)
+        
+        let screenChanged = hash1 != hash2
```

Comment:
Note that iOS may emit two screens that are equal, even during an animation.
To reduce the risk of early a premature trigger, we can do:
```
let period = request.period
let start = Date()
var screenshot = XCUIScreen.main.screenshot()
while (Date().durationSince(start) < period) {
  val compare = XCUIScreen.main.screenshot()
  if hash(screenshot) != hash(compare) { return false }
}
return true
```

Reply:
discussed in private